### DISCUSSION TITLE
Does docusaurus support the generation of doc for in-code-comment as the doxgen does ? Thx.

### DISCUSSION THREAD

**Discussion by Kiddinglife** [facebook/docusaurus#5368] [General] [answered]
docusaurus does nice jobs in generating techinical documentations from markdown file. but what about the in-code structured comments ?
Take the below c++comments as example used by doxgen tool:
```
/// \brief Initialize the, immediately setting the data it contains to a predefined pointer.
/// \details Set \a _copyData to true if you want to make an internal copy of the data you are passing. Set it to false to just save a pointer to the data.
/// You shouldn't call Write functions with \a _copyData as false, as this will write to unallocated memory
/// 99% of the time you will use this function to cast Packet::data for reading, in which case you should write something as follows:
/// \code
/// xxx::xxx(packet->data, packet->length, false);
/// \endcode
/// \param[in] _data An array of bytes.
/// \param[in] lengthInBytes Size of the \a _data.
/// \param[in] _copyData true or false to make a copy of \a _data or not.
xxx( unsigned char* _data, const unsigned int lengthInBytes, bool _copyData );
```

**Comment by slorber** [comment, marked as answer]
Hey

We don't provide first-class support for Doxygen. 

The docs plugin expects markdown files to be written to the file-system, so if you can get Doxygen to output md files, you could be able to integrate it with Docusaurus by generating those md files as a pre-build step. Some existing sites do this to generate reference API doc, but not surer anyone has done this with Doxygen (that I don't particularly know)

If Doxygen can output HTML files, you can also embed your Doxygen doc inside Docusaurus using iframes.

**Reply by Kiddinglife** [reply]
It looks a good workaround. But I am wondering whether the docusaurus themes work with those converted md/HTML files or not. Thank you.

**Reply by slorber** [reply]
>  But I am wondering whether the docusaurus themes work with those converted md/HTML files or not. Thank you.

It's your responsibility to emit md files that are compatible with Docusaurus/MDX in the first place.

Some examples:
- https://github.com/edno/graphql-markdown
- https://github.com/milesj/docusaurus-plugin-typedoc-api

**Reply by antonk52** [reply]
There is a [doxysaurus](https://www.npmjs.com/package/@rnw-scripts/doxysaurus) package that may be useful for integrating doxygen with docusaurus. I have not tried it, though
